cli/repair/patch.go

```diff
@@ -0,0 +1,169 @@
+	var resMessages []common.ResultMessage
+
+	currentConfContent, err := topologyConf.MarshalContent()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to marshal current content: %s", err)
+	}
+
+	if err := patchFunc(topologyConf, ctx); err != nil {
+		return nil, fmt.Errorf("Failed to patch topology config: %s", err)
+	}
+
+	newConfContent, err := topologyConf.MarshalContent()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get new config content: %s", err)
+	}
+
+	if ctx.Repair.DryRun || ctx.Cli.Verbose {
+		configDiff, err := getDiffLines(currentConfContent, newConfContent, "", "")
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get config difference: %s", err)
+		}
+
+		if len(configDiff) > 0 {
+			resMessages = append(resMessages, common.GetInfoMessage((strings.Join(configDiff, "\n") + "\n")))
+		} else {
+			resMessages = append(resMessages, common.GetInfoMessage("Topology config wasn't changed"))
+		}
+	}
+
+	return resMessages, nil
+}
+
+func rewriteConf(topologyConfPath string, topologyConf *TopologyConfType) ([]common.ResultMessage, error) {
+	var resMessages []common.ResultMessage
+
+	resMessages = append(resMessages, common.GetDebugMessage("Topology config file: %s", topologyConfPath))
+
+	backupPath, err := createFileBackup(topologyConfPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create topology config backup: %s", err)
+	}
+	resMessages = append(resMessages, common.GetDebugMessage("Created backup file: %s", backupPath))
+
+	newConfContent, err := topologyConf.MarshalContent()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get new config content: %s", err)
+	}
+
+	confFile, err := os.OpenFile(topologyConfPath, os.O_WRONLY|os.O_TRUNC, 0755)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open a new config: %s", err)
+	}
+
+	if _, err := confFile.Write(newConfContent); err != nil {
+		return nil, fmt.Errorf("Failed to write a new config: %s", err)
+	}
+
+	return resMessages, nil
+}
+
+func reloadConf(topologyConfPath string, instanceName string, ctx *context.Ctx) ([]common.ResultMessage, error) {
+	var resMessages []common.ResultMessage
+
+	consoleSock := project.GetInstanceConsoleSock(ctx, instanceName)
+	resMessages = append(resMessages, common.GetDebugMessage("Instance console socket: %s", consoleSock))
+
+	if _, err := os.Stat(consoleSock); err != nil {
+		return nil, fmt.Errorf("Failed to use instanace console socket: %s", err)
+	}
+
+	conn, err := net.Dial("unix", consoleSock)
+	if err != nil {
+		return resMessages, fmt.Errorf("Failed to dial: %s", err)
+	}
+
+	defer conn.Close()
+
+	// eval
+	evalFuncTmpl := `
+		local ClusterwideConfig = require('cartridge.clusterwide-config')
+		local confapplier = require('cartridge.confapplier')
+
+		local roles_configured_state = 'RolesConfigured'
+		local connecting_fullmesh_state = 'ConnectingFullmesh'
+
+		local state = confapplier.wish_state(roles_configured_state, {{ .WishStateTimeout }})
```

Comment:
It's better to wish_state right before apply_config.

Reply:
Fixed in #361